src/test/rustdoc/redirect-map.rs

```diff
@@ -0,0 +1,18 @@
+// compile-flags: -Z unstable-options --generate-redirect-map
+
+#![crate_name = "foo"]
+
+// @!has foo/private/struct.Quz.html
+// @!has foo/hidden/struct.Bar.html
+// @has foo/redirect-map.json
```

Comment:
Can you make this a `run-make` test so you can make sure the JSON is actually accurate? You'll need the changes from https://github.com/rust-lang/rust/pull/81197 so rustdoc is available.

Reply:
I actually wanted to ask about how to test the json. I'll keep this test and add another one in `run-make`.